platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceImpl.java

```diff
@@ -533,99 +535,91 @@ protected Application findFeature(Feature feature, Set<Application> applications
    * @param features
    * @return {@link BundleStateSet} containing information on the state of each bundle
    */
-  private final BundleStateSet getCurrentBundleStates(Set<Feature> features) {
+  private BundleStateSet getCurrentBundleStates(Set<Feature> features) {
     BundleStateSet bundleStateSet = new BundleStateSet();
     BundleContext context = getContext();
-
     for (Feature curFeature : features) {
       for (BundleInfo curBundleInfo : curFeature.getBundles()) {
+
         Bundle curBundle = context.getBundle(curBundleInfo.getLocation());
+        if (curBundle == null
+            || curBundle.adapt(BundleRevision.class).getTypes() == BundleRevision.TYPE_FRAGMENT) {
```

Comment:
❗️ `getTypes()` actually return an int which could potentially include multiple types 'ored' together. The proper condition should be: ```(curBundle.adapt(BundleRevision.class).getTypes() & BundleRevision.TYPE_FRAGMENT) == BundleRevision.TYPE_FRAGMENT```

Reply:
I'm not sure we need to consider OR'd types. The original condition would allow processing IFF: 

- `curBundle` is not `null`
- `getTypes()` is **not** equal to the fragment type

I'm fairly certain a proper DeMorgan's of that statement is to *skip* processing if *either*:

- `curBundle` is `null`
OR
- `getTypes()` is equal to the fragment type (without modification)